scripts/validate_metadata_conf.py

```diff
@@ -0,0 +1,188 @@
+    for section in SECTIONS:
+        keys = list(config[section].keys())
+        if section == 'metadata':
+            target = METADATA
+        elif section == 'data':
+            target = DATA
+        elif section == 'misc':
+            target = MISC
+
+        missing = [k for k in target if k not in keys]
+        unexpected = [k for k in keys if k not in target]
+        if missing or unexpected:
+            print(f"--> WARNING: [{section}] DOES NOT CONTAIN THE REQUIRED KEYS")
+            if missing:
+                print(f"    --> MISSING KEYS: {', '.join(missing)}")
+            if unexpected:
+                print(f"    --> UNEXPECTED KEYS: {', '.join(unexpected)}")
+            failures += 1
+
+    return failures
+
+
+def Validate_required_fields(config: configparser, failures: int) -> int:
+    """
+    Validates the contents of the required fields for a single .cfg file.
+
+        :param config: The config parser.
+        :param failures: The number of files that have failed validations.
+        :returns: The number of files that have failed validations.
+    """
+    # Verify that all required fields are not None
+    for section in SECTIONS:
+        for key, value in config[section].items():
+            if key in REQUIRED and value is None:
+                print(f"--> WARNING: {key} IS A REQUIRED FIELD AND CANNOT BE EMPTY")
+                failures +=1
+            if key == "branch_method" and value == "standard":
+                if key in PARENT_REQUIRED and value is None:
+                    print(f"--> WARNING: {key} IS A REQUIRED FIELD AND CANNOT BE EMPTY")
+                    failures += 1
+            if key == "mass_data_class":
+                if value == "ens":
+                    if key == "mass_ensemble_member" and not value:
+                        print(f"--> WARNING: {key} IS A REQUIRED FIELD AND CANNOT BE EMPTY")
+                        failures += 1
+                elif value == "crum":
+                    if key == "mass_ensemble_member" and value:
+                        print(f"--> WARNING: {key} IS NOT NEEDED WHEN USING CRUM MASS DATA CLASS")
+                        failures += 1
+
+    return failures
+
+
+def validate_field_inputs(config: configparser, failures: int) -> int:
+    """
+    Validates the inputs of a single .cfg file.
+
+        :param config: The config parser.
+        :param failures: The number of files that have failed validations.
+        :returns: The number of files that have failed validations.
+    """
+    for section in SECTIONS:
+        for key, value in config[section].items():
+            # Verify datetime inputs
+            if key == "branch_method" and value == "standard":
+                DATETIME_FIELDS.append("branch_date_in_child")
+                DATETIME_FIELDS.append("branch_date_in_parent")
+            format = r"^(?:\d{4})-(?:0[1-9]|1[0-2])-(?:0[1-9]|[12]\d|3[01])T(?:[01]\d|2[0-3]):[0-5]\d:[0-5]\dZ$"
```

Comment:
In this routine the format variable is a string then a regex object, then back and forth.

This makes it a little more tricky to identify what it is. Can you restructure this so that outside the for loops you construct a dictionary of compiled regex objects and pick them up as needed?